src/arrayMethodReduce.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result;
+    let start;
+
+    if (arguments.length === 2) {
+      result = initialValue;
+      start = 0;
+    } else {
```

Comment:
I think you can give values to the variables where you introduce. Then you will not need the else statement, which makes code harder to read.